### PR TITLE
fix: repair flaking test job of setup e2e test environment

### LINKED ISSUE #3667: Flaking test: setup e2e test environment
#### Which jobs are flaking:
e2e test

#### Which test(s) are flaking:
e2e test(setup e2e test environment)

- https://github.com/karmada-io/karmada/actions/runs/5241565482/jobs/9464064674?pr=3661  (#3661)
- https://github.com/karmada-io/karmada/actions/runs/5211054867/jobs/9403105260?pr=3656 (#3656)

#### Anything else we need to know:
```
Waiting for kubeconfig file /home/runner/.kube/members.config and clusters member1 to be ready...
Waiting for running............................................................................................................................................................................................................................................................................................................
Error:  Timeout waiting for condition running
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by chaosi-zju**
**What type of PR is this?**

/kind flake

**What this PR does / why we need it**:

for CI stability

**Which issue(s) this PR fixes**:

Fixes #3667 

**Special notes for your reviewer**:
@RainbowMango @XiShanYongYe-Chang @yike21 

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by chaosi-zju** [comment]
> Hi! @chaosi-zju Glad to see your contribution! Usually please use `git commit -s -m 'your message '` or `git commit -m ' Signed-off-by: AuthorName <authoremail@example.com> \n <other message> '` to pass DCO. And `rebase` to the latest codebase before committing is appreciated.
> 
> ```shell
> # an example
> # in branch `fake-ci`
> git fetch upstream
> git rebase upstream/master
> git commit -s --amend
> git push -f
> ```

thank you very much~

File: hack/util.sh [hunk 1]
```diff
@@ -426,6 +426,28 @@ function util::kubectl_with_retry() {
...
+  local member_config=${2}
+  local log_path=${3}
+
+  local log_file="${log_path}"/delete-all-clusters.log
+  rm -rf ${log_file}
+  mkdir -p ${log_path}
+
+  kind delete clusters --all >> "${log_file}" 2>&1
+  rm -f "${main_config}"
+  rm -f "${member_config}"
```
Comment (RainbowMango) [inline hack/util.sh:446]: It just feels a little bit rough to delete the entire kubeconfig file.
If the file contains a cluster not created by kind, the configuration will be removed unexpectedly. 

It's not a big deal though.

**Review by RainbowMango — COMMENTED** [review]
Generally looks good to me.
@yike21 do you have any other comments?

File: hack/util.sh (hunk 1 above)
Comment (yike21) [reply hack/util.sh:446]: Generally LGTM. It seems that the [logic for removing kubeconfig](https://github.com/karmada-io/karmada/blob/master/hack/local-down-karmada.sh#L37-L50) is included here. And the [logic for removing kind clusters](https://github.com/karmada-io/karmada/blob/master/hack/local-down-karmada.sh#L29-L35) scope is expanded. 

I think it's more suitable to just delete kind clusters which created by `local-up-karmada.sh`.

Comment (chaosi-zju) [reply hack/util.sh:446]: > It just feels a little bit rough to delete the entire kubeconfig file. If the file contains a cluster not created by kind, the configuration will be removed unexpectedly.
> 
> It's not a big deal though.

Although `kind delete` is capable of cleaning kubeconfig if specificed `$KUBECONFIG` or `--kubeconfig`, it is not as effective as expected. I have tried these two modification method:

1. one

```shell
export KUBECONFIG="${main_config}":"${member_config}"
kind delete clusters --all >> "${log_file}" 2>&1
unset KUBECONFIG
```

2. two

```shell
kind delete clusters --kubeconfig "${main_config}" karmada-host
kind delete clusters --kubeconfig "${member_config}" member1 member2 member3
```

All above methods can not clean kubeconfig content thoroughly.

***

By the way, even if no `rm -f xxx_config` in `delete_all_clusters`, following the original logic, the script will eventually execute `rm -f xxx_config` in `util::create_cluster`. 

the code of `util::create_cluster`:

```bash
function util::create_cluster() {
  local cluster_name=${1}
  local kubeconfig=${2}
  local kind_image=${3}
  local log_path=${4}
  local cluster_config=${5:-}

  mkdir -p ${log_path}
  rm -rf "${log_path}/${cluster_name}.log"
  rm -f "${kubeconfig}"
  ...
}
```

**Comment by yike21** [comment]
LGTM

File: hack/util.sh (hunk 1 above)
Comment (RainbowMango) [reply hack/util.sh:446]: OK. got it.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**